utilities/blob_db/blob_db_impl.cc

```diff
@@ -462,6 +480,83 @@ void BlobDBImpl::ProcessCompactionJobInfo(const CompactionJobInfo& info) {
       LinkSstToBlobFile(output.file_number, output.oldest_blob_file_number);
     }
   }
+
+  MarkUnreferencedBlobFilesObsolete();
```

Comment:
This function is outside any mutex protection.  Does it mean that we need to reason about correctness of multithreads calling it? Is it necessary? What would be the consequence if we guard it with some mutex?

Reply:
There are two variants of this method that rely on the same common private helper. The one called here, `MarkUnreferencedBlobFilesObsolete`, takes locks internally. The other one, `MarkUnreferencedBlobFilesObsoleteDuringOpen` is called only during `Open` and does not need synchronization.